### PR TITLE
feat: add support for points campaign

### PR DESCRIPTION
**Pull request by guerrap**

### CODE REVIEW

File: packages/frontend/src/commons/index.ts [hunk 1]
```diff
@@ -51,6 +55,11 @@ export interface ChainData {
 
 export const FEE_UNIT = 1_000_000;
 
+export const BLACKLISTED_ADDRESSES: Address[] = [
```
Comment (luzzif) [inline packages/frontend/src/commons/index.ts:58]: Why did we add this? I don't think it's used anywhere else.

File: packages/frontend/src/components/create-campaign/preview/index.tsx [hunk 2]
```diff
@@ -55,6 +61,72 @@ export function CampaignPreview({
     const publicClient = usePublicClient();
     const { writeContractAsync } = useWriteContract();
 
+    const tokensToApprove = useMemo(() => {
+        if (payload.rewardType === RewardType.tokens) return payload.tokens;
+
+        if (payload.rewardType === RewardType.points && payload.feeToken) {
+            const { amount, token } = payload.feeToken;
+
+            const newRaw = amount.raw + (amount.raw * 15n) / 100n;
```
Comment (luzzif) [inline packages/frontend/src/components/create-campaign/preview/index.tsx:70]: Slightly more efficient: amount.raw * 115n / 100

File: packages/frontend/src/commons/index.ts (hunk 1 above)
Comment (guerrap) [reply packages/frontend/src/commons/index.ts:58]: It was used in the distribution breakdown, before the changes, to hide the zero and dead addresses in the leaderboard, should we add this on the API level?

File: packages/frontend/src/components/create-campaign/preview/index.tsx (hunk 2 above)
Comment (guerrap) [reply packages/frontend/src/components/create-campaign/preview/index.tsx]: Addressed in b2e893253bb59bfceac8e3195fa15f23d58512e3.

File: packages/frontend/src/commons/index.ts (hunk 1 above)
Comment (luzzif) [reply packages/frontend/src/commons/index.ts:58]: We already do with the 0 address. It's fine if the dead address shows up, as it indicates an error in the backend distribution. I think we can remove this

Comment (guerrap) [reply packages/frontend/src/commons/index.ts]: Addressed in 9b157fb30b55f767c13f77f37040bdf92dd7a7f2.

**Pull request merged**